src/puptoo/app.py

```diff
@@ -201,9 +201,7 @@ def handle_message(msg):
 
     if msg.get("service") == "advisor":
         facts = process_archive(msg, extra)
-    if msg.get("service") == "compliance":
-        facts = msg.get("metadata")
-    if msg.get("service") == "malware-detection":
+    if msg.get("service") in ["compliance", "malware-detection"]:
```

Comment:
Since you are consuming the new announce topic, it might be worth looking at the kafka headers, rather than opening up the message and having to process the JSON to get to this key. The JSON processing takes roughly twice as long as reading the `service` header from kafka